src/components/route/booklist.jsx

```diff
@@ -1,35 +1,37 @@
-import React from 'react';
-import { useSelector, useDispatch } from 'react-redux';
-import { removeBook } from '../../redux/books/booksSlice';
-import '../css/bookStyle.css';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { removeBook, fetchData } from '../../redux/books/booksSlice';
 
 const BookList = () => {
-  const books = useSelector((state) => state.books.booksItem);
   const dispatch = useDispatch();
+  const books = useSelector((state) => state.books);
+  useEffect(() => {
+    dispatch(fetchData());
+  }, [dispatch]);
   return (
     <div>
-      <h3>Show books</h3>
-      <ul>
-        {books.map((book) => (
-          <li key={book.item_id}>
-            <p>
-              Title:
-              {book.title}
-            </p>
-            <p>
-              Author:
-              {book.author}
-            </p>
-            <p>
-              Category:
-              {book.category}
-            </p>
-            <button type="button" onClick={() => dispatch(removeBook({ item_id: book.item_id }))}>Remove</button>
-          </li>
-        ))}
-      </ul>
-    </div>
+      {books.booksItem.map((book) => (
+        <div key={book.key}>
+          <h4>
```

Comment:
![image](https://github.com/AliShahSafdari/bookstore/assets/93039359/d9fd9f3d-e4c9-41c9-af12-c91fd92df334)

- [ ] As you can see in the image above that we are facing some issue in the console 😨 which is regarding the key value of the each book so it would be nice if you give a unique key to each book to not get this error in the future because it can break your whole website 🤔

You can make your lines 13 and 14 as below to make your project not face the above error 👍

```JavaScript
      {books.booksItem.map((book, index) => (
        <div key={index}>
```

Reply:
Thanks, I solve that problem.